clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1061,25 +1069,29 @@ private RequestFuture<Void> doCommitOffsetsAsync(final Map<TopicPartition, Offse
         future.addListener(new RequestFutureListener<Void>() {
             @Override
             public void onSuccess(Void value) {
-                inFlightAsyncCommits.decrementAndGet();
-
-                if (interceptors != null)
-                    interceptors.onCommit(offsets);
-                completedOffsetCommits.add(new OffsetCommitCompletion(cb, offsets, null));
+                try {
+                    if (interceptors != null)
+                        interceptors.onCommit(offsets);
+                    completedOffsetCommits.add(new OffsetCommitCompletion(cb, offsets, null));
+                } finally {
+                    inFlightAsyncCommits.decrementAndGet();
+                }
             }
 
             @Override
             public void onFailure(RuntimeException e) {
-                inFlightAsyncCommits.decrementAndGet();
+                try {
+                    Exception commitException = e;
 
-                Exception commitException = e;
-
-                if (e instanceof RetriableException) {
-                    commitException = new RetriableCommitFailedException(e);
-                }
-                completedOffsetCommits.add(new OffsetCommitCompletion(cb, offsets, commitException));
-                if (commitException instanceof FencedInstanceIdException) {
-                    asyncCommitFenced.set(true);
+                    if (e instanceof RetriableException) {
+                        commitException = new RetriableCommitFailedException(e);
+                    }
+                    completedOffsetCommits.add(new OffsetCommitCompletion(cb, offsets, commitException));
+                    if (commitException instanceof FencedInstanceIdException) {
```

Comment:
I know you didn't write the original logic, but it seems funny to compare the exception against a `FencedInstanceIdException` here if up above `commitException` was converted to a `RetriableCommitFailedException`. I guess there's not a better way to construct this without potentially changing the logic to be incorrect.

Reply:
`FencedInstanceIdException` is not retriable, so this shouldn't make a difference, right?